src/model.ts

```diff
@@ -157,27 +162,45 @@ export class JSORMBase {
   private static _credentialStorage?: CredentialStorage
 
   static get credentialStorage(): CredentialStorage {
-    if (!this._credentialStorage) {
-      if (!this._credentialStorageBackend) {
-        if (this.jwtStorage && typeof localStorage !== "undefined") {
-          this._credentialStorageBackend = localStorage
-        } else {
-          this._credentialStorageBackend = new NullStorageBackend()
-        }
-      }
-
-      this._credentialStorage = new CredentialStorage(
-        this.jwtStorage,
-        this._credentialStorageBackend
-      )
+    if (this.baseClass === undefined) {
+      throw new Error(`No base class for ${this.name}`)
     }
 
-    return this._credentialStorage
+    if (!this._credentialStorage) {
+      return this._initializeCredentialStorage()
+    } else {
+      return this._credentialStorage
+    }
   }
 
   static set credentialStorageBackend(backend: StorageBackend | undefined) {
+    if (this.baseClass === JSORMBase) {
```

Comment:
I am only choosing to do this because it was the way to preserve as much of the previous behavior as possible while allowing us to fix this functionality.  If we are fine with users being able to set a base credential that must be manually overridden at lower levels (instead of setting one for each base class by default), then I'm fine with that as well, it will just result in a couple more breaking changes.